<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
  <meta http-equiv="Content-Language" content="en-us">
  <meta http-equiv="Content-Type"
 content="text/html; charset=windows-1252">
  <title>DAWG-PAWS: Manual</title>
  <link rel="stylesheet" type="text/css" href="css/dawgpaws.css">
  <link rel="SHORTCUT ICON" href="images/dawgpaws.ico">
</head>
<body>
<table style="border-collapse: collapse;" id="AutoNumber1" border="0"
 bordercolor="#111111" cellpadding="0" cellspacing="0" width="100%">
  <tbody>
    <tr>
      <td width="50%">
      <p align="center"> <img src="images/DAWGPAWS_med.png"
 align="left" border="2" height="115" width="326"></p>
      </td>
      <td width="50%">
      <ul>
      </ul>
      <h1>DAWG-PAWS<br>
User Manual</h1>
      <a href="http://jestill.myweb.uga.edu">James C. Estill</a><br>
Nov. 10, 2008<br>
      <br>
      <h2><span style="text-decoration: underline;"><span
 style="font-style: italic;"></span></span></h2>
      <ul>
      </ul>
      </td>
    </tr>
  </tbody>
</table>
<hr>
<h1><span style="color: rgb(255, 0, 0);">!! THIS DOCUMENT IS UNDER
CURRENT
DEVELOPMENT !!<br>
</span></h1>
<h4>This program manual and the scripts that make up the DAWG-PAWS
package are under current development. Everything is subject to change
without notice at this point. This software comes as is, without any
expressed or implied warranty. Use at your own risk.<br>
</h4>
<h1>TABLE OF CONTENTS<br>
</h1>
<h2><a style="font-weight: bold;" href="#getting_started">I. GETTING
STARTED<br>
</a></h2>
<div style="margin-left: 40px;">
<ol style="list-style-type: upper-alpha;">
  <li><a style="font-weight: bold;" href="#dawg_paws_installation">Install
DAWG-PAWS</a></li>
  <li><a href="#install_additional_software"><span
 style="font-weight: bold;">Install Required Software</span></a></li>
  <li><a href="#install_perl_modules"><span style="font-weight: bold;">Install
Required Perl Modules</span></a></li>
  <li><a href="#define_env_vars"><span style="font-weight: bold;">Define
Environment Variables</span></a></li>
  <li><span style="font-weight: bold;"><a href="#install_test">Test
Installation</a><br>
    </span></li>
</ol>
</div>
<h2><a href="#annotation_process">II.
ANNOTATION PROCESS</a><br>
</h2>
<div style="margin-left: 40px;">
<ol style="list-style-type: upper-alpha;">
  <li><a style="font-weight: bold;" href="#sequence_preparation">Prepare
Sequence Files for Annotation</a></li>
  <ol>
    <li><a href="#prep_split_multifasta">Split FASTA Files with
multiple records</a><br>
    </li>
    <li><a href="#prep_fasta_shorten">Shorten FASTA file names</a></li>
    <li><a href="#prep_soft_mask">Softmask Sequence files</a></li>
    <li><a href="#prep_hard_mask">Hardmask Sequence files</a><br>
    </li>
  </ol>
  <li><a style="font-weight: bold;" href="#structural_features">Structural
Feature Annotation</a></li>
  <ol>
    <li><a href="#run_struc_annotate_gaps">Annotate Gaps in the Assembly</a><br>
    </li>
  </ol>
  <li><a style="font-weight: bold;" href="#gene_annotation">Gene
Annotation De Novo Computational Results</a></li>
  <ol start="1">
    <li><a href="#run_genscan">GenScan</a><br>
    </li>
    <li><a href="#run_genemark">GeneMark.hmm</a></li>
    <li><a href="#run_eugene">EuGene</a></li>
    <li><a href="#run_geneid">GeneID</a><br>
    </li>
    <li><a href="#run_fgenesh">FGenesh</a></li>
  </ol>
  <li><a style="font-weight: bold;" href="#te_annotation">Transposable
Element Computational Results</a></li>
  <ol>
    <li><a href="#run_te_repeatmasker">RepeatMasker</a><br>
    </li>
    <li><a href="#ltr_struc">LTR_Struc</a></li>
    <li><a href="#ltr_finder">LTR_FINDER</a></li>
    <li><a href="#find_ltr">find_ltr</a></li>
    <li><a href="#ltr_seq">ltr_seq</a><a href="man.html#run_genscan">GENSCAN</a></li>
    <li><a href="#te_nest">TE Nest</a></li>
    <li><a href="#findmite">FINDMITE</a></li>
    <li><a href="#oligo_count">Oligomer Counts</a></li>
    <li><a href="#hmmer_TE">HMMER - TE Models</a></li>
    <li><a href="#te_tandem_repeats_finder">Tandem Repeats Finder</a></li>
  </ol>
  <li>BLAST Based Homology Searches</li>
  <ol>
    <li><a href="man.html#run_ncbi_blast">NCBI-BLAST</a></li>
  </ol>
  <li><a style="font-weight: bold;" href="#apollo_preparation">Preparing
Computational Results for Apollo</a></li>
  <li><a style="font-weight: bold;" href="#human_curation">Human
Curation of Computational Results</a></li>
  <li><a style="font-weight: bold;" href="#curation_quality_control">Curation
Quality Contro</a><a style="font-weight: bold;"
 href="#curation_quality_control">l</a></li>
  <ol>
    <li>Convert game.xml annotations to GFF<br>
    </li>
    <li>Venn Diagrams of Annotated Sequence Features</li>
    <li>Gene Model HMMAlignments of predicted genes<br>
    </li>
  </ol>
</ol>
</div>
<h3><a href="#additional_information">III.
ADDITIONAL INFORMATION</a></h3>
<div style="margin-left: 40px;">
<ol>
  <li><a style="font-weight: bold;" href="#apollo_data_tiers">Apollo
Data Tiers</a></li>
  <li><a style="font-weight: bold;" href="#directory_structure">Computational
Results Directory Structure</a></li>
</ol>
</div>
<h3><a style="font-weight: bold;" href="#manfiles">IV. MANUAL FILES FOR
INDIVUDAL PROGRAMS</a></h3>
<h3><a href="#literature_cited">V. LITERATURE CITED</a><br>
</h3>
<br>
<hr style="width: 100%; height: 2px;">
<h1><a name="getting_started"></a>I. GETTING STARTED<br>
</h1>
Getting started using DAWG-PAWS may require that you install some the
required software or define some variables in your user environment.
The steps involved in preparing to do genome annotation with the
DAWG-PAWS package is described in detail below.<br>
<br>
This user documentation will generally assume that you are running
DAWG-PAWS on a Unix or Linux machine and that you are operating from
the <a href="http://www.gnu.org/software/bash/">bash shell</a>.
Anywhere where I have written <big><code>yourname</code></big> below,
simply substitute your user name.<br>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="dawg_paws_installation"></a>Install DAWG-PAWS</h1>
Currently the best way to download the DAWG-PAWS program is to
anonymously check
out the SVN code from the code repository on SourceForge. This will
require that you have the SVN client program installed which can be
downloaded from <a
 href="http://subversion.tigris.org/project_packages.html">http://subversion.tigris.org/project_packages.html</a>.
Once you have the SVN client program installed, navigate to the
location where you want to place the DAWG-PAWS program and check out
the
code. This is illustrated by the following commands:<br>
<pre class="prettyprint"><span class="str">  &gt;mkdir /home/yourname/apps<br>  &gt;cd /home/yourname/apps/<br>  &gt;svn co https://dawgpaws.svn.sourceforge.net/svnroot/dawgpaws/trunk dawgpaws </span><span
 class="typ"></span><span class="pun"></span><span class="pln"></span><span
 class="pun"></span><span class="pln"></span><span class="pun"></span><span
 class="pln"></span><span class="pun"></span></pre>
The above command will download all the working copies of the scripts,
program manuals, and emacs modes from the SourceForge repository. These
will be placed in the following
directory structure:<br>
<ul>
  <li style="font-weight: bold;"><span style="font-weight: normal;"><span
 style="font-weight: bold;">dawgpaws</span><br>
This base directory will be created when you check out dawgpaws for the
first time<br>
    </span></li>
  <ul>
    <li><span style="font-weight: bold;">apollo</span><br>
      <span style="font-style: italic;">The tiers and type file for
using the apollo program are installed here</span><br>
    </li>
    <li><span style="font-weight: bold;">docs</span><br>
      <span style="font-style: italic;">Additional documentation for
using the DAWG-PAWS program and the full text of the GNU GPL license.</span><br>
    </li>
    <li><span style="font-weight: bold;">emacs_modes</span><br>
      <span style="font-style: italic;">Major modes files for emacs,
this will provide syntax highlighting to the apollo tiers file, types
files and configuration files.</span><br>
    </li>
    <li><span style="font-weight: bold;">html</span><br>
      <span style="font-style: italic;">This is a copy of the web pages
that are hosted at http://dawgpaws.sourceforge.net</span><br>
    </li>
    <ul>
      <li><span style="font-weight: bold;">htdocs</span><br>
        <span style="font-style: italic;">The base directory of the
html files. This includes the pages that are posted at the SourceForge
web site.</span><br>
      </li>
      <li><span style="font-weight: bold;">cgi-bin</span><br>
        <span style="font-style: italic;">Programs run through the
dawgpaws web are here.</span><br>
      </li>
    </ul>
    <li><span style="font-weight: bold;">scripts</span><br>
      <span style="font-style: italic;">The scripts that are the
backbone of DAWG-PAWS are stored in this directory</span></li>
    <li><span style="font-weight: bold;">t</span><span
 style="font-style: italic;"><span style="font-style: italic;"><br>
      </span>Test files for checking if DAWG-PAWS is ready to go<br>
      </span></li>
    <ul>
      <li><span style="font-weight: bold;">data</span><span
 style="font-style: italic;"><br>
Data files that will be used by the test scripts.<br>
        </span></li>
    </ul>
  </ul>
</ul>
The svn co command will check out the version of DAWG-PAWS that is the
most up to date at the time of your initial download. To update to the
most recent version of DAWG-PAWS at any time in the future you will
need to run the svn update command from within the dawgpaws
directory:<br>
<pre class="prettyprint"><span class="str">  &gt;cd /home/yourname/apps/dawgpaws<br>  &gt;svn update </span></pre>
This will update to the most current version of DAWG-PAWS.<br>
<h3>Symbolic Link to the Perl Binary</h3>
In Linux and the Mac OS X environment, it is possible to run all of the
DAWG-PAWS programs without needing to type <big><code>perl</code></big>
before the
program name. To make this
possible, all of the DAWG-PAWS programs assume that your local
installation of perl is at <big><code>/usr/bin/perl</code></big>. You
can see this by opening
up a DAWG-PAWS program in a text editor. You will see that <big><code>#!/usr/bin/perl
-w</code> </big>is the first line in the program.<br>
<br>
If your local copy of the perl binary is not at the <big><code>/usr/bin/perl</code></big>
location, you can make a symbolic link using the ln command. In
general, the ln command is used as <br>
<pre class="prettyprint"><span class="str">  &gt;ln existing_file new_name<br></span></pre>
where existing_file is the name of the file on your machine, and
new_name is the name to the new shortcut. The following command example
shows to to make a shortcut when your local installation of perl is at
/usr/local/bin/perl instead of the expected /usr/bin/perl. <br>
<pre class="prettyprint"><span class="str">  &gt;ln -s /usr/local/bin/perl /usr/bin/perl<br></span></pre>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="install_additional_software"></a>Install Additional
Required Software</h1>
The DAWG-PAWS suite of programs are scripts that provide for high
throughput execution of existing genome annotation software. The
following represents an alphabetical list of software and web services
that may be used by DAWG-PAWS. This list provides a link to the source
files for installation of the software or links to the web services.
The Operating Systems that the software can be executed on is also
listed as well as references to the peer reviewed publications
describing
the software. You do not need
to install all of these programs to make use of the DAWG-PAWS
package. This is just the complete list of programs that DAWG-PAWS can
use.<br>
<ul>
  <li style="font-weight: bold;">Apollo : Genome Annotation Curation
Tool<br>
  </li>
  <ul>
    <li>Source: <a href="http://www.fruitfly.org/annot/apollo/">http://www.fruitfly.org/annot/apollo/</a><br>
    </li>
    <li>OS: Win32, Linux32, Mac OS X</li>
    <li>Ref: Lewis, S. E., S. M. Searle, et al. (2002). Genome Biol
3(12): RESEARCH0082.<br>
    </li>
  </ul>
  <li><span style="font-weight: bold;">BLAST (NCBI-BLAST)</span> <br>
  </li>
  <ul>
    <li>Source <a
 href="http://www.ncbi.nlm.nih.gov/blast/download.shtml">http://www.ncbi.nlm.nih.gov/blast/download.shtml</a></li>
    <li>OS: Win32, Win64, Linux32, Linux64, Mac OS X, Solaris32,
Solaris64<br>
    </li>
    <li>Ref: Altschul, S. F., T. L. Madden, et al. (1997). Nucleic
Acids Res 25(17): 3389-402.</li>
  </ul>
  <li><span style="font-weight: bold;">Cross_Match</span><span
 style="font-style: italic;"><br>
Cross_match is part of the Phrap package and is used by RepeatMasker.
It is a general purpose application for comparing any two DNA sequence
sets.</span></li>
  <ul>
    <li>Source: <a href="http://www.phrap.org/">http://www.phrap.org/</a><br>
    </li>
  </ul>
  <li><span style="font-weight: bold;">EuG</span><span
 style="font-weight: bold;">è</span><span style="font-weight: bold;">ne<br>
    </span><i>An
open gene finder for eukaryotic organisms. Compared to most existing
gene finders, EuGène is characterized by its ability to simply
integrate arbitrary sources of information in its prediction process.</i></li>
  <ul>
    <li>Source:<a
 href="http://www.inra.fr/internet/Departements/MIA/T/EuGene/"
 rel="nofollow">http://www.inra.fr/internet/Departements/MIA/T/EuGene/</a>
    </li>
    <li>Ref: Schiex, T., A. Moisan, et al. (2001). Computational
Biology. 111-125.<br>
    </li>
  </ul>
  <li style="font-weight: bold;">find_ltr <span
 style="font-style: italic; font-weight: normal;"><br>
The find_ltr program is a de novo LTR Retrotransposon discovery tool</span><br>
  </li>
  <ul>
    <li>Source:<a
 href="http://darwin.informatics.indiana.edu/evolution/LTR.tar.gz">http://darwin.informatics.indiana.edu/evolution/LTR.tar.gz</a></li>
    <li>Ref:
Rho, M., J. H. Choi, et al. (2007). BMC Genomics 8: 90.<br>
    </li>
  </ul>
  <li style="font-weight: bold;">FINDMITE<span
 style="font-weight: normal;"></span><i style="font-weight: normal;"><br>
Identifies putative MITES based on
structural criteria.</i></li>
  <ul>
    <li>Source:<a href="http://jaketu.biochem.vt.edu/dl_software.htm"
 rel="nofollow">http://jaketu.biochem.vt.edu/dl_software.htm</a> </li>
    <li>OS: Redhat Linux<br>
    </li>
    <li>Ref: Tu, Z. (2001). Proc Natl Acad Sci U S A 98(4): 1699-704.<br>
    </li>
  </ul>
  <li style="font-weight: bold;">FGENESH<span
 style="font-weight: normal;"></span><br>
  </li>
  <ul>
    <li>Source:<a href="http://www.softberry.com/berry.phtml"
 rel="nofollow">http://www.softberry.com/berry.phtml</a></li>
  </ul>
  <li style="font-weight: bold;">GeneID<span
 style="font-weight: normal;"></span><br>
  </li>
  <ul>
    <li>Source:<a href="http://www1.imim.es/software/geneid/index.html"
 rel="nofollow">http://www1.imim.es/software/geneid/index.html</a></li>
    <li>Ref: Parra, G., E. Blanco, et al. (2000). Genome Res 10(4):
511-5.<br>
    </li>
  </ul>
  <li style="font-weight: bold;">GeneMarkHMM</li>
  <ul>
    <li>Source:<a href="http://opal.biology.gatech.edu/GeneMark/"
 rel="nofollow">http://opal.biology.gatech.edu/GeneMark/</a></li>
    <li>Ref: Lukashin, A. V. and M. Borodovsky (1998). Nucleic Acids
Res 26(4): 1107-15.<br>
    </li>
  </ul>
  <li style="font-weight: bold;">GENSCAN</li>
  <ul>
    <li>Source:<a href="http://genes.mit.edu/GENSCANinfo.html"
 rel="nofollow">http://genes.mit.edu/GENSCANinfo.html</a></li>
    <li>Ref: Burge, C. and S. Karlin (1997). J Mol Biol 268(1): 78-94.<br>
    </li>
  </ul>
  <li><span style="font-weight: bold;">HMMER<br>
    </span><span style="font-style: italic;">HMMER is a freely
distributable implementation of profile HMM software for protein
sequence analysis. </span><br>
  </li>
  <ul>
    <li>Source:<a href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a></li>
    <li>OS: Linux32,Linux64,Mac OS X</li>
  </ul>
  <li><span style="font-weight: bold;">LTR_FINDER</span> <br>
  </li>
  <ul>
    <li>Source :A Linux Binary is available upon contacting the
authors : xuzh &lt;at&gt; fudan.edu.cn</li>
    <li>Web Query:<a href="http://tlife.fudan.edu.cn/ltr_finder/">http://tlife.fudan.edu.cn/ltr_finder/</a></li>
    <li>Ref:
Xu, Z. and H. Wang (2007). Nucleic Acids Res
35(Web Server issue): W265-8.</li>
  </ul>
  <li><span style="font-weight: bold;">LTR_Seq</span> - <i>Identifies
LTR retrotransposons
based on structural criteria</i></li>
  <ul>
    <li>Source:<a href="http://www.eecs.wsu.edu/%7Eananth/software.htm"
 rel="nofollow">http://www.eecs.wsu.edu/~ananth/software.htm</a></li>
    <li>OS: Linux<br>
    </li>
    <li>Ref: Kalyanaraman, A. and S. Aluru (2006).&nbsp; J Bioinform
Comput Biol 4(2): 197-216.<br>
    </li>
  </ul>
  <li style="font-weight: bold;">LTR_Struc</li>
  <ul>
    <li>Source:<a
 href="http://www.genetics.uga.edu/retrolab/data/LTR_Struc.html">http://www.genetics.uga.edu/retrolab/data/LTR_Struc.html</a></li>
    <li>OS: Win32<br>
    </li>
    <li>Ref: McCarthy, E. M. and J. F. McDonald (2003). Bioinformatics
19(3): 362-7.</li>
  </ul>
  <li style="font-weight: bold;">RepeatMasker<span
 style="font-weight: normal;"></span><i style="font-weight: normal;"><br>
RepeatMasker is a program
that screens DNA sequences for interspersed repeats and low complexity
DNA sequences</i></li>
  <ul>
    <li>Source:<a href="http://www.repeatmasker.org/" rel="nofollow">http://www.repeatmasker.org/</a></li>
  </ul>
  <li style="font-weight: bold;">Tandem Repeats Finder<span
 style="font-weight: normal;"> <br>
    </span><font style="font-weight: normal; font-style: italic;"
 size="-1">Locates and displays tandem repeats in DNA sequences.</font></li>
  <ul>
    <li>Source:<a href="http://tandem.bu.edu/trf/trf.html"
 rel="nofollow">http://tandem.bu.edu/trf/trf.download.html</a></li>
    <li>OS: Win32, Linux32, Linux64, Mac OSX, Solaris<br>
    </li>
    <li>Ref: Benson, G. (1999). Nucleic Acids Res 27(2): 573-80.<br>
    </li>
  </ul>
  <li style="font-weight: bold;">TE Nest <span
 style="font-weight: normal;"><br>
    </span><i style="font-weight: normal;">TE
nest is an annotation and visualization tool for the identification
transposable elements. TE nest is unique in that it reconstructs
transposable elements separated by nesting of subsequent insertions.</i></li>
  <ul>
    <li>Web Query:<a
 href="http://www.plantgdb.org/prj/TE_nest/TE_nest.html" rel="nofollow">http://www.plantgdb.org/prj/TE_nest/TE_nest.html</a></li>
    <li>Ref: Kronmiller, B. A. and R. P. Wise (2007). Plant Physiol. in
press<br>
    </li>
  </ul>
  <li style="font-weight: bold;">TriAnnot <span
 style="font-weight: normal;"><span style="font-style: italic;"><br>
The purpose the TriAnnot web server is to provide a web based automated
annotation system for wheat similar to existing resources for rice.</span></span><br>
  </li>
  <ul>
    <li>Web Query:<a
 href="http://urgi.versailles.inra.fr/projects/TriAnnot/pipeline.php">http://urgi.versailles.inra.fr/projects/TriAnnot/pipeline.php</a><br>
    </li>
  </ul>
  <li style="font-weight: bold;">Vmatch<span
 style="font-style: italic; font-weight: normal;"><br>
A tool for efficiently
solving large scale sequence matching tasks using a persistent index. </span><i
 style="font-style: italic; font-weight: normal;">Vmatch</i><span
 style="font-style: italic; font-weight: normal;"> subsumes the
software Reputer (Kurtz et al 2001).</span> </li>
  <ul>
    <li>Source:<a href="http://www.vmatch.de/">http://www.vmatch.de/</a></li>
    <li>OS: Linux32, Linux64, Mac OSX, Solaris32, Solaris64</li>
    <li>Program Manual : <a
 href="http://www.zbh.uni-hamburg.de/vmatch/virtman.pdf">http://www.zbh.uni-hamburg.de/vmatch/virtman.pdf</a><br>
    </li>
  </ul>
  <li><span style="font-weight: bold;">WUBLAST</span><span
 style="font-style: italic;"><br>
WUBLAST is a Local Alignment program similar to NCBI-BLAST but it quite
different in its implementation , speed and results.</span><br>
  </li>
  <ul>
    <li>Source:<a href="http://blast.wustl.edu/licensing/">http://blast.wustl.edu/licensing/</a></li>
    <li>OS: Linux32, Linux64, Mac OSX, Solaris32, Solaris64</li>
  </ul>
</ul>
<br>
<hr style="width: 100%; height: 2px;">
<h1><a name="install_perl_modules"></a>Install Required Perl Modules<br>
</h1>
The following Perl modules are required:<br>
<ul>
  <li><span style="font-weight: bold;">BioPerl</span><span
 style="font-style: italic;"><br>
BioPerl is used on a few scripts. </span><br>
  </li>
  <ul>
    <li>Source: <a
 href="http://www.bioperl.org/wiki/Installing_BioPerl">http://www.bioperl.org/wiki/Installing_BioPerl</a><br>
    </li>
  </ul>
</ul>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="define_env_vars"></a>Define Environment Variables</h1>
<br>
In the Unix-like operating systems including Unix, Linux, and Mac OS X,
you can define variables in your command line environment. These
variables change the way you interact with the command line, and can be
used to simplify command tasks. Modifying the environment will
allow you to use a program at the command line without referring to its
full path, and can simplify using DAWG-PAWS scripts by letting you
define common variables in your environment instead of defining them at
the
command line every time you use a program. The following information
shows you how to define environment variables for use by the DAWG-PAWS
suite of programs. For more information on the shell and user
environment in Linux you can refer to <a
 href="http://www.comptechdoc.org/os/linux/usersguide/linux_ugenvironment.html">http://www.comptechdoc.org/os/linux/usersguide/linux_ugenvironment.html</a><br>
<h4>Add the DAWG-PAWS code directory</h4>
Adding the DAWG-PAWS code directory to your path will allow you type in
the DAWG-PAWS commands without needing to type the entire path to the
DAWG-PAWS program. For example, you will be able to just type <big><code>batch_blast.pl</code></big>
to launch the batch_blast program.pl instead of needing to type <big><code>/home/yourname/code/dawgpaws/scripts/batch_blast.pl</code></big>.
In the bash shell you would add the following line to your user profile.<br>
<pre class="prettyprint"><span class="str">  export PATH=$PATH:$HOME/apps/dawgpaws/scripts</span></pre>
<h4>Define paths to required software</h4>
Many of the DAWG-PAWS command line programs use external software that
may be stored at different locations in different user environments.
You can define the location of these external programs at the command
line, but you are also able to define these paths in your user
environment. <br>
<br>
An example of settings these path variables in the bash shell is below:<br>
<br>
<pre class="prettyprint"><span class="str">1.  <span
 style="color: rgb(255, 0, 0);">#FIND_LTR</span><br></span>2.  export PATH=$PATH:/home/yourname/apps/LTRDeNovo/LTR/tool<br><br></pre>
The above information is not complete.<br>
<h4>DAWG-PAWS Options</h4>
A number of environment variable options are also available for the
DAWG-PAWS scripts. These variables include things such as the path to
commonly used programs, paths to&nbsp; user name and dsn connection for
logging onto databases, a<br>
<h4>Full variable set for the bash shell</h4>
Putting the above set of environment options together in a single file
gives the following. In the bash shell you can copy and paste the
following to your .bashrc file. You will of course need to modify the
directory paths to the true locations of the software on your machine.<br>
<big><big><span style="font-weight: bold;"></span></big></big><br>
<pre class="prettyprint"><span class="str"><span
 style="color: rgb(255, 0, 0);">#-----------------------------+</span></span><br><span
 class="str"><span style="color: rgb(255, 0, 0);"># SOFTWARE                    |<br></span></span><span
 class="str"><span style="color: rgb(255, 0, 0);">#-----------------------------+<br><br># DAWG-PAWS SCRIPTS <br></span></span><span
 class="str">export PATH=$PATH:$HOME/code/dawgpaws/scripts</span><br><br><span
 style="color: rgb(255, 0, 0);"># FIND_LTR</span><br>export PATH=$PATH:$HOME/apps/LTRDeNovo/LTR/tool<br>export FIND_LTR_ROOT='/home/yourname/apps/LTRDeNovo/LTR/tool/'<span
 class="str"><span style="color: rgb(255, 0, 0);"><br><br># LTR_FINDER</span></span><br><span
 class="str">export TRNA_DB='$HOME/apps/LTR_FINDER.'<br>export PROSITE_DIR='$HOME/apps/LTR_FINDER/ps_scan/'<br><br></span><span
 class="str"><span style="color: rgb(255, 0, 0);"># LTR_seq</span></span><br><span
 class="str">export PATH=$PATH:$HOME/apps/ltr_seq<br><br></span><span
 class="str"><span style="color: rgb(255, 0, 0);"># EuGene</span></span><br><span
 class="str">export EUGENEDIR='$HOME/apps/eugene-3.3'</span><br>&nbsp;<span
 class="str"><span style="color: rgb(255, 0, 0);"><br>#-----------------------------+</span></span><br><span
 class="str"><span style="color: rgb(255, 0, 0);"># DAWG-PAWS VARS              |<br></span></span><span
 class="str"><span style="color: rgb(255, 0, 0);">#-----------------------------+<br># The following variables are used directly by DAWG-PAWS scripts<br><br></span></span><span
 class="str"><span style="color: rgb(255, 0, 0);"></span></span><span
 style="color: rgb(255, 0, 0);"># LTR_Finder path</span><br><span
 class="str">export LTR_FINDER='$HOME/apps/ltr_finder/ltr_finder'</span>
<br><span class="str"><span style="color: rgb(255, 0, 0);"># GENSCAN <br></span></span><span
 class="str">export DP_GENSCAN_BIN='$HOME/apps/genscan/genscan'</span><br>export DP_GENSCAN_LIB='$HOME/apps/genscan/Maize.smat'<br><br><span
 class="str"><span style="color: rgb(255, 0, 0);"># GeneMark</span>                                <br>export GM_BIN_DIR='$HOME/apps/GenMark/genemark_hmm_euk.linux/'<br>export GM_LIB_DIR='$HOME/apps/GenMark/genemark_hmm_euk.linux/'<br><br><span
 style="color: rgb(255, 0, 0);"># VennMaster</span><br>export VMASTER_DIR='/$HOME/apps/VennMaster/VennMaster-0.36.0/'<br>export VMASTER_JAVA_BIN='/usr/java/jre1.6.0/bin/java'<br></span></pre>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="install_test"></a>Test Your Installation of DAWG-PAWS<br>
</h1>
<br>
I am currently writing test scripts to test your DAWG::PAWS
installation.<br>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="annotation_process"></a>II. ANNOTATION PROCESS</h1>
Despite the name, the use of DAWG-PAWS for annotation is really more of
a process then a pipeline. There is not a single place where you drop
in a sequence, push a button and get a fully annotated genome as
output. The series of scripts that comprise DAWG-PAWS are designed to
be robust across operating systems as much as possible, and are
expected to be usable in an
<span style="font-style: italic;">a la carte </span>fashion. The
following outline of an annotation process
indicates the use of the full suite DAWG-PAWS programs; however, a
subset of these programs
could be chosen for use in producing computational results for genome
annotation.<br>
<br>
The process described below assumes that you will be curating the
computational evidences using the Apollo Genome Annotation Curation
tool. I assume that you are using the game.xml file format as the
working copy of you annotation.<br>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="sequence_preparation"></a>A. Preparing Sequence Files for
Annotation<br>
</h1>
The following is
an overview of the process used to annotate sequences with the
DAWG-PAWS set of programs. This assumes that you are running the
analysis on a Linux machine, and that you are storing your annotation
computational results in the directory
<pre class="prettyprint"><span class="str">  /home/</span><span
 class="typ">yourname</span><span class="pun">/</span><span class="pln">projects</span><span
 class="pun">/</span><span class="pln">wheat_annotation</span><span
 class="pun">/</span><span class="pln">wheat_analysis</span><span
 class="pun">/</span></pre>
<p>This
directory will have a subdirectory named for every annotated sequence.
The subdirectories for the annotated sequence will follow the directory
structure indicated in the <a href="man.html#directory_structure">DAWG-PAWS
Directory Structure</a> below. </p>
<p>In all of the following command examples, the &gt; character
represents the command line prompt. </p>
<h1><a name="prep_split_multifasta"></a>1. Split MutliFASTA Files</h1>
The entire DAWG-PAWS process assumes that each FASTA file contains a
single record representing a large contig such as a B. If your query
sequence file contains multiple fasta
files, you will first need to split the fasta file into individual
records.<br>
<h3><br>
</h3>
<h1><a name="prep_fasta_shorten"></a>2. Rename FASTA files to a short
unique name</h1>
<p>Because
many of the programs that are used in the DAWG-PAWS process have limits
on the size of FASTA headers, it may be necessary to first shorten the
name of the FASTA files. </p>
<p>Starting with a directory in which you have a single fasta file for
every BAC to process at the following directory path </p>
<pre class="prettyprint"><span class="str">  /home/</span><span
 class="typ">YourName</span><span class="pun">/</span><span class="pln">projects</span><span
 class="pun">/</span><span class="pln">wheat_annotation</span><span
 class="pun">/</span><span class="pln">wheat_analysis</span><span
 class="pun">/</span><span class="pln">fasta_orig</span></pre>
<p>Navigate to the wheat_analysis directory using the cd command to
change directories: </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation</span><span class="pun">/</span><span
 class="pln">wheat_analysis</span></pre>
<p>Issue the following command from the wheat_analysis directory: </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">fasta_shorten</span><span
 class="pun">.</span><span class="pln">pl </span><span class="pun">-</span><span
 class="pln">i fasta_orig</span><span class="pun">/</span><span
 class="pln"> </span><span class="pun">-</span><span class="pln">o fasta_short</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">l </span><span class="lit">10</span><span class="pln"> </span><span
 class="pun">--</span><span class="pln">uppercase</span></pre>
<p>This
will rename all of the fasta files in the fasta_orig directory and will
place the results in the fasta_short directory. The -l option of 10
will shorten the names to 10 characters, and the --uppercase flag will
convert all lowercase base calls to uppercase. </p>
<h1><a name="prep_soft_mask"></a>3. Soft mask the renamed files using
RepeatMasker and the TREP
database</h1>
<p>For the wheat BACs, the <a
 href="http://wheat.pw.usda.gov/ITMI/Repeats/flatfile.nr" rel="nofollow">nonredundant
TREP database</a> is used. </p>
<p>Navigate to the wheat_annotation directory </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation</span></pre>
<p>Issue the following command from the wheat_annotation directory </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">batch_mask</span><span
 class="pun">.</span><span class="pln">pl </span><span class="pun">-</span><span
 class="pln">i wheat_analysis</span><span class="pun">/</span><span
 class="pln">fasta_short</span><span class="pun">/</span><span
 class="pln"> </span><span class="pun">-</span><span class="pln">o wheat_analysis</span><span
 class="pun">/</span><span class="pln"> <br>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span><span
 class="pun">-</span><span class="pln">c wheat_analysis</span><span
 class="pun">/</span><span class="pln">batch_mask</span><span
 class="pun">.</span><span class="pln">jcfg <br>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span><span
 class="pun">--</span><span class="pln">engine wublast</span></pre>
<p>This will softmask all fasta files in the fasta_short directory and
will place the results in the wheat_analysis directory. </p>
<p>The
-c option should point to the location of the batch_mask configuration
file. The --engine wublast option will use the wublast engine for
masking the sequences. The default behavior is to mask with crossmatch.
</p>
<h1><a name="prep_hard_mask"></a>4. Hard mask the softmasked files
generated above</h1>
<p>Many
of the gene prediction programs are not soft-masked "aware" so it is
necessary to also make a hard masked copy of the fasta file. This will
replace the lowercase letters with an N or X character. The example
below shows masking the fasta files in the soft_masker directory with
an uppercase N and placing the output in the hard_masked directory.<br>
</p>
<pre class="prettyprint"><span class="pln"> </span><span class="pun"> &gt;</span><span
 class="pln">cd </span><span class="pun">/</span><span class="pln">home</span><span
 class="pun">/</span><span class="pln">YourName</span><span class="pun">/</span><span
 class="pln">projects</span><span class="pun">/</span><span class="pln">wheat_annotation</span><span
 class="pun">/</span><span class="pln">wheat_analysis</span><span
 class="pln"> &nbsp;</span><span class="pun"><br>  &gt;</span><span
 class="pln">batch_hardmask</span><span class="pun">.</span><span
 class="pln">pl </span><span class="pun">-</span><span class="pln">i soft_masked</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o hard_masked</span><span class="pun">/ -m N</span></pre>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="structural_features"></a>B. Structural Feature Annotation<br>
</h1>
DAWG-PAWS can currently only annotate a few structural features.
Structural features are defined here as features of the sequence such
as gaps, gc content, etc. that are not directly related to the&nbsp;
annotation of biological sequence features such as genes or
transposable elements.<br>
<h1><a name="run_struc_annotate_gaps"></a>1. Annotate the gaps in the
assembly</h1>
<p>The following will find gaps in your fasta files.<br>
</p>
<pre class="prettyprint"><span class="pln">&nbsp; </span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation</span><span class="pun">/</span><span
 class="pln">wheat_analysis<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">batch_findgaps</span><span class="pun">.</span><span
 class="pln">pl </span><span class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">masked_soft</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/</span><span
 class="pln"></span></pre>
The batch_findgaps.pl program is slow and uses and ugly algorithm but
it does work. Results will be placed in gff format in the gff directory
as well as in game.xml format in the game directory.<br>
<br>
<hr style="width: 100%; height: 2px;">
<h1><a name="gene_annotation"></a>C. Gene Annotation
De Novo Computational Results</h1>
The information below assumes that you have properly prepared the
sequence files for annotation <a href="#sequence_preparation">as
described above</a>. You can currently use the following five De Novo
gene annotation programs within DAWG-PAWS. The native results from
these programs are translated to a gff format that can be use with the
Apollo program.<br>
<br>
<h1><a name="run_genscan"></a>1. GENSCAN gene prediction program</h1>
Required Programs:<br>
<ul>
  <li>GENSCAN : <a href="http://genes.mit.edu/GENSCANinfo.html"
 rel="nofollow">http://genes.mit.edu/GENSCANinfo.html</a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_genscan.pl">batch_genscan.pl</a><br>
  </li>
</ul>
The GENSCAN program is not part of the TriAnnot web sevice, thus the
GENSCAN program must be run on your local machine. It is also possible
to run the GENSCAN program on a web server such as <a
 href="http://genes.mit.edu/GENSCAN.html">http://genes.mit.edu/GENSCAN.html</a>.
If you do Genscan
on a remote web server, you may be able to convert the results to GFF
format using the program cnv_genscan2gff.pl.<br>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">batch_genscan</span><span
 class="pun">.</span><span class="pln">pl </span><span class="pun">-</span><span
 class="pln">i wheat_analysis</span><span class="pun">/</span><span
 class="pln">hard_masked</span><span class="pun">/</span><span
 class="pln"> </span><span class="pun">-</span><span class="pln">o wheat_analysis</span><span
 class="pun">/</span></pre>
<h1><a name="run_genemark"></a>2. GenMarkHMM<br>
</h1>
Required Programs:<br>
<ul>
  <li>GeneMarkHMM : <a href="http://opal.biology.gatech.edu/GeneMark/"
 rel="nofollow">http://opal.biology.gatech.edu/GeneMark/</a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_genemark.pl">batch_genemark.pl</a><br>
  </li>
</ul>
The batch_genemark.pl program allows you to run GeneMark
locally. The GenMarkHMM program requires that you obtain a license to
run it on your local machine.<br>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">batch_genemark</span><span
 class="pun">.</span><span class="pln">pl </span><span class="pun">-</span><span
 class="pln">i wheat_analysis</span><span class="pun">/</span><span
 class="pln">hard_masked</span><span class="pun">/</span><span
 class="pln"> </span><span class="pun">-</span><span class="pln">o wheat_analysis/</span><span
 class="pun"></span></pre>
<h1>3. FGeneSH</h1>
Required Programs:<br>
<ul>
  <li>FGeneSH : <a href="http://www.softberry.com/berry.phtml"
 rel="nofollow">http://www.softberry.com/berry.phtml</a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/cnv_fgenesh2gff.pl?revision=309">cnv_fgenesh2gff.pl</a></li>
</ul>
The FGeneSH program can be run in a limited form on the Softberry web
site. Since a license for this program is so expensive to rent, I have
not had access to the code to write a script for running this program
in batch mode. An alternative to running this program on the softberry
web server is to use the TriAnnot web service.<br>
<h1><a name="run_geneid"></a>4. GeneID<br>
</h1>
Required Programs:<br>
<ul>
  <li>geneid : <a href="http://www1.imim.es/software/geneid/index.html"
 rel="nofollow">http://www1.imim.es/software/geneid/index.html</a></li>
  <li>batch_geneid.pl<br>
  </li>
</ul>
To run the geneid program in batch mode you would use the following
command. <br>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;batch_geneid.pl </span><span class="pln"></span><span
 class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">hard_masked</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/ --param wheat.param.Apr_22_2004</span></pre>
You may also specify the full path to the geneid binary using the the
--geneid-path option.<br>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;batch_geneid.pl </span><span class="pln"></span><span
 class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">hard_masked</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/ --param wheat.param.Apr_22_2004<br>                   --geneid-path /usr/local/geneid.March_1_2005<br></span></pre>
The original results from the geneid program will be placed in the
directory named geneid/ while the gff formatted results prepared for
Apollo will be placed in the gff/ directory.<br>
<h1><a name="run_eugene"></a>5. EuGène<br>
</h1>
Required Programs:<br>
<ul>
  <li>EuGène : <a
 href="http://www.inra.fr/internet/Departements/MIA/T/EuGene/"
 rel="nofollow">http://www.inra.fr/internet/Departements/MIA/T/EuGene/</a></li>
  <li>batch_eugene.pl</li>
</ul>
Running the EuGène program in batch mode will make use of the hard
masked input fasta files, as well as a parameter file created for your
organism. Parameter files in EuGène allow you to specify many<br>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;batch_eugene.pl </span><span class="pln"></span><span
 class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">hard_masked</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/ --param wheat_eugene.par</span></pre>
The HTML and images files resulting from EuGene will be placed in the
eugene directory. The original gff output from EuGene will also be
placed in the eugene directory, while a gff file translated for use in
Apollo will be placed in the gff direcotry. The results from this
batch_eugene run will differ from the results on the TriAnnot web
server due to differences in the paramter files between the two
pipelines.<br>
<br>
<hr><br>
<h1><a name="te_annotation"></a>D. Transposable Element Computational
Results<br>
</h1>
The information below assumes that you have properly prepared the
sequence files for annotation <a href="man.html#sequence_preparation">as
described above</a>. The following transposable element annotation
steps can generally be done in any order, and you may choose to not
include some of these programs in your analysis pipeline.<br>
<h1><a name="run_te_repeatmasker"></a>1. RepeatMask Sequences<br>
</h1>
Required Programs:<br>
<ul>
  <li>RepeatMasker : <a href="http://www.repeatmasker.org/"
 rel="nofollow">http://www.repeatmasker.org/</a></li>
</ul>
The process for running RepeatMasker to identify known Transposable
Elements is described above.<br>
<h2>
</h2>
<h1><a name="ltr_struc"></a>2. LTR_Struc Program for LTR
Retrotransposon Prediction<br>
</h1>
Required Programs:<br>
<ul>
  <li>LTR_Struc : <a
 href="http://www.genetics.uga.edu/retrolab/data/LTR_Struc.html">http://www.genetics.uga.edu/retrolab/data/LTR_Struc.html</a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/ltrstruc_prep.pl">ltrstruc_prep.pl<br>
    </a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_ltrstruc.vbs">batch_ltrstruc.vbs</a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/cnv_ltrstruc2gff.pl">cnv_ltrstruc2gff.pl<br>
    </a></li>
</ul>
The LTR_Struc program has not been optimized for use in a
high-throughput fashion, it has a executable binary that is only
available for the Windows platform, and it does not provide output in a
form that can be easilty mapped back onto the query sequence. The
source code for LTR_Struc is not available to allow for modifications
of these limiting factors. The process for running LTR_Struc therefore
requires some extraneous steps as outlined below:<br>
<ol>
  <li>Prepare sequence files for analysis in the MS Windows
environment. The <big><code>ltrstruc_prep.pl </code></big>program
will convert the UNIX format line endings to DOS formated files with
the *.txt extension and will also create the flist.txt file required by
LTR_Struc.<br>
    <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">ltrstruc_prep</span><span class="pun">.</span><span
 class="pln">pl </span><span class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">masked_soft</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o for_ltrstruc</span><span class="pun">/</span><span
 class="pln"></span><span class="pun"></span><span class="pln"></span><span
 class="pun"></span><span class="pln"></span></pre>
  </li>
  <li>Copy the directory of transformed fasta files and flist.txt to a
MS Windows machine for analysis and place these in the same directory
as the LTR_Struc binary.<br>
  </li>
  <li>Use the <big><code>batch_ltrstruc.vbs</code></big> program to
run the LTR_Struc program in batch mode. This program is a visual basic
script program that works to send the information to the LTR_Struc
command line. This has been tested and known to work under the Windows
XP operating system. You can <a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_ltrstruc.vbs">download
batch_ltrstruc.vbs here</a> from the DAWG-PAWS subversion repository
and place this in the same directory as the ltr_struc binary. As
written, this will run LTR_Struc under the most stringent conditions.<br>
    <pre class="prettyprint"><span class="pln"> &nbsp;C:</span><span
 class="pun">&gt;cd ltrstrucdir<br>  C:&gt;Cscript.exe //NoLogo batch_ltrstruc.vbs | cmd.exe<br></span></pre>
  </li>
  <li>When the LTR_Struc analysis is complete, transfer the LTR_Struc
output back to your Linux machine.<br>
  </li>
  <li>Use the <big><code>cnv_ltrstruc2gff.pl</code></big> program to
convert the LTR_Struc output to a format that can be mapped back onto
the query sequence. This script will extract sequence strings as
reported by LTR_Struc and will map the strings back onto the query
sequence using a Perl based string matching function. This matching
function currently assumes only a single match to this string exists in
the query sequence.<br>
    <pre class="prettyprint"><span class="pln">&nbsp;</span><span
 class="pun"> &gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">cnv_ltrstruc2gff.pl -i wheat_analysis/masked_soft/ -o wheat_analysis/<br>                       --results wheat_analysis/ltrstruc_results/</span><span
 class="pln"></span></pre>
  </li>
</ol>
<h1><a name="ltr_finder"></a>3. LTR_FINDER Program for LTR Structure
Prediction<br>
</h1>
Required Programs:<br>
<ul>
  <li>LTR_FINDER</li>
  <ul>
    <li>LTR_FINDER Web : <a
 href="http://tlife.fudan.edu.cn/ltr_finder/">http://tlife.fudan.edu.cn/ltr_finder/</a><br>
    </li>
    <li>LTR_FINDER Manual : <a
 href="http://tlife.fudan.edu.cn/ltr_finder/help/help.pdf">http://tlife.fudan.edu.cn/ltr_finder/help/help.pdf</a></li>
    <li>Binary : A Linux Binary is available upon contacting the
authors : xuzh &lt;at&gt; fudan.edu.cn</li>
  </ul>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_ltrfinder.pl">batch_ltrfinder.pl<br>
    </a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_ltrfinder.pl">cnv_ltrfinder2gff.pl</a></li>
</ul>
The following assumes that you have obtained a binary of the LTR_FINDER
program. It is also possible to generate LTR_FINDER Results using the <a
 href="http://tlife.fudan.edu.cn/ltr_finder/">LTR_FINDER web page</a>,
but I have not tested the DAWG-PAWS scripts on the web page output. It
is possible that the cnv_ltrfidner2gff.pl program will work for single
fasta files analyzed with the LTR_FINDER web page.<br>
<br>
The LTR_FINDER program has been designed with high throughput use in
mind, it provides rich output for the location and biology of the LTR
retrotransposons that it predicts, and it allows for fine scale control
of search parameters. The batch_ltrfinder.pl program provides an
interface to run LTR_FINDER for multiple parameter sets for each query
sequence, and it converts output to the GFF file format.<br>
<pre class="prettyprint"><span class="pln">&nbsp;</span><span
 class="pun"> &gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>  &gt;batch_ltrfinder.pl -i wheat_analysis/masked_soft/ -o wheat_analysis/<br>                      -c config/batch_ltrfinder.jcfg<br></span></pre>
<h1><a name="find_ltr"></a>4. Run the find_ltr program</h1>
Required Programs:<br>
<ul>
  <li>find_ltr.pl : This is a component of the LTRDeNovo packge<br>
This is available at : <a
 href="http://darwin.informatics.indiana.edu/evolution/LTR.tar.gz">http://darwin.informatics.indiana.edu/evolution/LTR.tar.gz</a><br>
  </li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_findltr.pl">batch_findltr.pl</a><br>
  </li>
</ul>
It is unclear if the current license for the find_ltr program allows
modification of the source code.&nbsp; If the license prevents
modification of the program source, this would unfortunately include
modifying paramaters that are important to using this program for LTR
discovery. I have modified the program code to accept these parameter
changes at the command line, and the batch_findltr.pl program depends
on these source code modifications. An example of using the find_ltr
program in batch mode is illustrated below:<br>
<pre class="prettyprint"><span class="pln">&nbsp;</span><span
 class="pun"> &gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>  &gt;batch_findltr.pl -i wheat_analysis/masked_soft -o wheat_analysis<br>                    -c config/batch_findltr.jcfg<br></span></pre>
The above example will produce a GFF file for each parameter set in the
batch_findltr.jcfg file for each fasta input sequence recognized in the
masked_soft input directory.<br>
<br>
<h1><a name="ltr_seq"></a>5. Run the ltr_seq program</h1>
Required Programs:<br>
<ul>
  <li>ltr_seq : this is the latest incarnation of the ltr_par program<br>
A binary of this program is available upon contacting the program
author:<br>
    <a href="http://www.eecs.wsu.edu/%7Eananth/contact.htm">http://www.eecs.wsu.edu/~ananth/contact.htm</a><br>
  </li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/cnv_ltrseq2gff.pl">cnv_ltrseq2gff.pl</a><br>
  </li>
</ul>
Running ltr_seq in batch mode is currently not supported. You must do
an individual run of ltr_seq manually and then convert the output to
gff using the <a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/cnv_ltrseq2gff.pl">cnv_ltrseq2gff.pl</a>
program.<br>
<br>
<pre class="prettyprint"><span class="pln">&nbsp;</span><span
 class="pun"> &gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation/wheat_analysis/HEX0014K09<br>  &gt;mkdir ltr_seq<br>  &gt;RUN LTR_SEQ PROGRAM<br>&nbsp; </span><span
 class="pun">&gt;</span><span class="pln">cnv_ltrseq2gff.pl -i ltr_seq/ltr_seq_out.txt -o gff/HEX0014K09_ltrseq.gff<br>                     -s HEX0014K09<br></span></pre>
This will produce a file name HEX0014K09_ltrseq.gff that contains all
of the output from the LTR_seq program that was determined to be an LTR
Retrotransposon. The current version of the LTR_seq program appears to
produce duplicate predictions.<br>
<h1><a name="te_nest"></a>6. TE Nest analysis for LTR Retrotransposon
Nesting<br>
</h1>
Required Programs:<br>
<ul>
  <li>TE Nest : <a
 href="http://www.public.iastate.edu/%7Eimagefpc/Subpages/software.html">http://www.public.iastate.edu/~imagefpc/Subpages/software.html</a></li>
  <ul>
    <li>The TE Nest program is currently available on the web from
PlantGDB:<br>
      <a href="http://www.plantgdb.org/prj/TE_nest/TE_nest.html">http://www.plantgdb.org/prj/TE_nest/TE_nest.html</a></li>
  </ul>
  <li><a href="man/fetch_tenest.html">fetch_tenest.pl</a></li>
</ul>
The TE Nest program is provided as a <a
 href="http://www.public.iastate.edu/%7Eimagefpc/Subpages/software.html">web
page based service</a> from the Plant GDB web server. The TE Nest
program uses a homology based search approach to
identify nested elements in your query sequence. The fetch_tenest.pl
program can be used to bulk download results from the TE Nest web
server and convert results from the TE Nest text format to the GFF
format.<br>
<ol>
  <li>Submit your sequence jobs to the TE Nest web server <a
 href="http://www.plantgdb.org/PlantGDB-cgi/TE_nest/cgi/displayTE.pl">http://www.plantgdb.org/PlantGDB-cgi/TE_nest/cgi/displayTE.pl</a></li>
  <li>Record the job id that is returned for your sequence file by
adding a row to your config file that indicates (1) The name of your
query sequence as used in the rest of your analysis, (2) the job id
that TE Nest assigned to your submision. An example config file is
available from the DAWG-PAWS SVN site.<br>
  </li>
  <li>When all of you jobs have completed on the TE Nest web server,
download the results using the fetch_tenest.pl program.<br>
    <pre class="prettyprint"><span class="pln">&nbsp;</span><span
 class="pun"> &gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">fetch_tenest.pl -c tenest_results.txt -o wheat_analysis/ --gff</span></pre>
This will place the results in the wheat_analysis folder with a
separate set of folders for query sequence defined in the config file.</li>
</ol>
An alternative is to run TE nest locally using the batch_tenest.pl
program.<br>
<br>
Required Programs:<br>
<ul>
  <li>TEnest : <span style="text-decoration: underline;"></span></li>
  <li><span style="text-decoration: underline;">batch_tenest.pl</span></li>
</ul>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">batch_tenest</span><span class="pun">.</span><span
 class="pln">pl </span><span class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">masked_soft</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/ --org wheat </span><span
 class="pln"></span><span class="pln"></span></pre>
TEnest will use the maize database by default. This script currently
does not convert the output to gff format. Valid options for --org
include barley, wheat, maize, rice
<h1><a name="findmite"></a>7. FINDMITE for MITE Discovery<br>
</h1>
Required Programs:<br>
<ul>
  <li>FINDMITE : <a href="http://jaketu.biochem.vt.edu/dl_software.htm"
 rel="nofollow">http://jaketu.biochem.vt.edu/dl_software.htm</a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_findmite.pl">batch_findmite.pl</a></li>
</ul>
The batch_findmite program will do a FINDMITE analysis for each
parameter
set in your configuration file for each query sequence in your input
directory. The results from FINDMITE have a VERY high false positive
rate so you will need to further evaluate your results to find the true
MITEs
in your query sequence.<br>
<br>
To run the batch_findmite.pl program you would do the following:<br>
<pre>  <span class="pun">&gt;</span><span class="pln">cd </span><span
 class="pun">/</span><span class="pln">home</span><span class="pun">/</span><span
 class="pln">YourName</span><span class="pun">/</span><span class="pln">projects</span><span
 class="pun">/</span><span class="pln">wheat_annotation</span>  <br>  &gt;batch_findminte.pl <span
 class="pln">-i wheat_analysis/masked_soft/ -o wheat_analysis/<br>                      -c config/batch_findmite.jcfg --gff<br></span></pre>
The FINDMITE results will be place in the findmite directory while the
results converted to the GFF format would be placed in the gff
directory.<br>
<h1><a name="oligo_count"></a>8. Oligomer Counts<br>
</h1>
Required Programs:<br>
<ul>
  <li>Vmatch : <a href="http://www.vmatch.de/">http://www.vmatch.de/<br>
    </a>The Vmatch package is available for noncomercial academic use
upon signing <a
 href="http://www.zbh.uni-hamburg.de/vmatch/Vmatchlic.pdf">a license
agreement form</a></li>
  <ul>
    <li><a href="http://www.zbh.uni-hamburg.de/vmatch/virtman.pdf">Vmatch
manual</a><br>
    </li>
  </ul>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/seq_oligocount.pl">seq_oligocount.pl<br>
    </a></li>
</ul>
The <big><code>seq_oligocount.pl</code></big> program is at an early
stage of development. It
currently breaks the query sequence into oligomers of length k, it then
uses the vmatch program to query these kmers against a index database
to determine oligo copy number depth in the index database. This index
database could be an index of the raw shotgun reads, an index of the
assembled reads, or a database index of an external dataset. The result
produced by <big><code>seq_oligocount.pl </code></big>is a GFF format
file that gives the
oligomer copy number of every segment of length k in the query
sequence. It would
make sense to allow for binning these results over some window,
but that is currently not implemented. The process for using this
program is outlined below.<br>
<ol>
  <li>Create a persistent index of your database with the mkvtree
program. The example below shows how to index the file named
my_seqs.fasta.<br>
    <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>  &gt;mkdir seq_index<br>  &gt;cp my_seqs.fasta seq_index/<br>  &gt;cd seq_index/<br>  &gt;mkvtree -db my_seqs.fasta -dna -allout -pl<br></span></pre>
This index can be used to generate an index of coverage.<br>
  </li>
  <li>Use the persistent index created above as a database to query
your sequence against using the seq_oligocount.pl program. The example
below shows how to create an oligo count tier for the fasta file
HEX0014K09.fasta where the length of the oligos is 20 bases and the
index file is the my_seqs.fasta file from above. The output will be
placed in the directory: wheat_analysis/HEX0014K09/<br>
    <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>  &gt;seq_oligocount.pl --infile masked_soft/HEX0014K09.fasta -n HEX0014K09<br>                     --db seq_index/my_seqs.fasta -k 20<br>                     --outdir wheat_analysis/HEX0014K09/</span></pre>
  </li>
</ol>
The seq_oligocount.pl program will place the GFF formatted results in
the gff directory.<br>
<h1><a name="hmmer_TE"></a>9.HMMER - TE Models</h1>
Required Programs:<br>
<ul>
  <li>hmmsearch : <a href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a><br>
  </li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/batch_hmmer.pl">batch_hmmer.pl</a></li>
  <li>Example Config file : <a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/config/batch_hmmer.jcfg">batch_hmmer.jcfg</a><br>
  </li>
</ul>
It is possible to use profile hidden Markov models of known
transposable elements to find the location of TEs in your query
sequence. This approach was used in to identiTfy MITES and MULEs in
the Rice genome (Juretic et al. 2004). You can produce your own Profile
HMMs
using multiple
alignments of sequences of your choice, or you can test this approach
using HMMs developed for rice.<br>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>  &gt;batch_hmmer.pl </span><span
 class="pln">-i wheat_analysis/masked_soft -o wheat_analysis<br>                  -c config/batch_hmmer.jcfg --gff</span>
</pre>
<h1><a name="te_tandem_repeats_finder"></a>10. Tandem Repeats Finder
(TRF)<br>
</h1>
Required Programs:<br>
<ul>
  <li>trf : <a href="http://tandem.bu.edu/trf/trf.download.html">http://tandem.bu.edu/trf/trf.download.html</a></li>
  <li>batch_trf.pl<span style="text-decoration: underline;"></span><br>
  </li>
</ul>
The Tandem Repeats Finder can be run in batch mode. This script will
work with TRF 4.0, I have not tried to use it with earlier versions of
TRF.<br>
<pre class="prettyprint"><span class="pln">  &gt;batch_trf.pl </span><span
 class="pln">-i wheat_analysis/masked_soft -o wheat_analysis<br>                -c config/batch_hmmer.jcfg</span></pre>
The TRF data file will be placed in the directory, TRF. The gff
formatted results will be placed in the gff directory.<br>
<br>
<hr style="width: 100%; height: 2px;">
<h1>E. NCBI-BLAST Homology Searches<br>
</h1>
Additional gene and TE annotation tiers are added using the NCBI BLAST
program.<br>
<h1><a name="run_ncbi_blast"></a>1. NCBI-BLAST processes</h1>
<p>Homology searches with NCBI-BLAST will use the soft masked files
generated above.<br>
</p>
<p>First, you will need to prepare sequence files to serve as databases
for BLAST queries. This will use the formatdb command. The DAWG-PAWS
program uses the name that BLAST assigns to your database, so you
should indicate the database name and ID when formatting the database.
You should choose a shortened form of the database name with no spaces.
The following shows how to format the fasta file
my_database_of_repeats.fasta using the small name repdb as the database
name.<br>
</p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;formatdb -i my_database_of_repeats.fasta -p F -n repdb -t repdb</span><span
 class="pln"></span></pre>
<p> </p>
<p>For a small number of blast jobs, you can run the BLAST jobs on your
machine using the batch_blast.pl program with the <a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/*checkout*/dawgpaws/trunk/scripts/config/batch_blast_full.jcfg"
 rel="nofollow">batch_blast_full.jcfg</a> configuration file. The -d
argument is used to indicate the location of the BLAST databses. </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">batch_blast</span><span
 class="pun">.</span><span class="pln">pl </span><span class="pun">-</span><span
 class="pln">i wheat_annotation</span><span class="pun">/</span><span
 class="pln">soft_masked </span><span class="pun">-</span><span
 class="pln">o wheat_annotation</span><span class="pun">/</span><span
 class="pln"> <br>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span><span
 class="pun">-</span><span class="pln">c batch_blast_full</span><span
 class="pun">.</span><span class="pln">jcfg </span><span class="pun">-</span><span
 class="pln">d </span><span class="pun">/</span><span class="pln">db</span><span
 class="pun">/</span><span class="pln">paws</span><span class="pun">/</span><span
 class="pln"> <br>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span><span
 class="pun">--</span><span class="pln">logfile wheat_annotation</span><span
 class="pun">/</span><span class="pln">blast_job</span><span class="pun">.</span><span
 class="pln">log</span></pre>
<p>For
larger number of BLAST jobs, the BLAST program will generally be run in
a parallel cluster computing framework. This will require the
following: </p>
<ol>
  <li>Split the fasta directory into subdirectories. Each subdirectory
will be analyzed on a separate node on the cluster machine. </li>
  <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">fasta_dirsplit </span><span
 class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">soft_masked </span><span
 class="pun">-</span><span class="pln">o wheat_analysis</span><span
 class="pun">/</span><span class="pln">blast_dirs</span><span
 class="pun">/</span><span class="pln"><br>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </span><span
 class="pun">-</span><span class="pln">n </span><span class="lit">16</span><span
 class="pln"> </span><span class="pun">-</span><span class="pln">b geterdone</span></pre>
  <li>Copy the new dirs to the cluster machine </li>
  <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">scp </span><span class="pun">-</span><span
 class="pln">p </span><span class="pun">-</span><span class="pln">r geterdone</span><span
 class="pun">*</span><span class="str">/ YourName@yourclustermachine:/</span><span
 class="pln">scratch</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">wheat_in</span><span class="pun">/</span></pre>
  <li>Run
blast processes on the cluster machine. The specifics of running the
blast processes on your cluster will depend on the schedulding software
used by your cluster. The easiest way to deal with this is to write a
shell script which submits the jobs to your queue. The you will just
need to run the
shell script to execute this blast on the cluster. </li>
  <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;.</span><span class="str">/subjob.sh</span></pre>
  <li>Download BLAST results to your local machine </li>
  <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">scp </span><span class="pun">-</span><span
 class="pln">p </span><span class="pun">-</span><span class="pln">r YourName@yourclustermachine</span><span
 class="pun">:</span><span class="str">/scratch/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">wheat_out</span><span class="pun">/</span><span
 class="pln"> <br>&nbsp; &nbsp; &nbsp; &nbsp;</span><span class="str">/home/</span><span
 class="pln">YourName</span><span class="pun">/</span><span class="pln">projects</span><span
 class="pun">/</span><span class="pln">wheat_annotation</span><span
 class="pun">/</span><span class="pln">blast_results</span></pre>
  <li>Merge the BLAST results into the wheat_analysis directory </li>
  <pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">dir_merge</span><span class="pun">.</span><span class="pln">pl </span><span
 class="pun">-</span><span class="pln">i blast_results</span><span
 class="pun">/</span><span class="pln">wheat_out </span><span
 class="pun">-</span><span class="pln">o wheat_analysis</span></pre>
</ol>
<p style="font-weight: bold;">Convert BLAST Results to GFF</p>
<p>All
of the BLAST results for the query sequence will be added to a single
gff output file named root_name_all_blast.gff. Where root_name is the
name of the sequence file being processed. </p>
<p>For example, the
file HEX0014K09.fasta will have the BLAST results named
HEX0014K09_all_blast.gff. The gff file will be placed in the directory <big><tt>wheat_analysis/HEX0014K09/gff</tt></big>.
The gff directory will be created if it does not already exist. </p>
<p>Currently, all blast files must end in a .blo, .bln,
or .blx file extension to be recognized as a BLAST output
file.</p>
<p>The program batch_cnv_blast2gff.pl provides the ability to convert
the blast output from </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">batch_cnv_blast2gff</span><span class="pun">.</span><span
 class="pln">pl &nbsp;-i wheat_analysis</span><span class="pun">/</span><span
 class="pln">soft_masked </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/</span><span
 class="pln"> </span></pre>
This assumes that the blast output is the full alignment format. If you
did the blast with the -m 8 alignment option, you will need to specify
this with hte -b option as shown below<br>
<pre class="prettyprint"><span class="pln"></span><span class="pun">&gt;</span><span
 class="pln">batch_cnv_blast2gff</span><span class="pun">.</span><span
 class="pln">pl &nbsp;-i wheat_analysis</span><span class="pun">/</span><span
 class="pln">soft_masked </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/ -b 8</span><span
 class="pln"> </span></pre>
<br>
<hr style="width: 100%; height: 2px;">
<h1><a name="apollo_preparation"></a> F. Preparing Computational
Results
for Apollo<br>
</h1>
<h2>1. Audit the computational results</h2>
<p>The audit program will move gff files to the gff dir if needed and
will alert
you to any expected files that could not be found. Currently this
program only audits a subset of the expected results from DAWG-PAWS
with a focus on gene annotation output. This program will probably end
up getting deprecated, but I will leave it here for now.<br>
</p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">batch_audit</span><span class="pun">.</span><span
 class="pln">pl </span><span class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">masked_soft</span><span
 class="pun">/</span><span class="pln"> </span><span class="pun">-</span><span
 class="pln">o wheat_analysis</span><span class="pun">/</span><span
 class="pln"> </span><span class="pun">--</span><span class="pln">full </span><span
 class="pun">--</span><span class="pln">color</span></pre>
<p>The <tt>--full</tt> option will run a complete audit. This
currently includes the TriAnnotation output, </p>
<p>The <tt>--color</tt> option will print error messages about missing
files in red font. </p>
<h2>2. Concatenate the gff files</h2>
<p>This is currently a manual step that needs to be automated. </p>
<p>For each query sequence, navigate to the directory containing the
gff output files. For example, for the file HEX0014K09 </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation</span><span class="pun">/</span><span
 class="pln">wheat_analysis</span><span class="pun">/</span><span
 class="pln">HEX0014K09</span><span class="pun">/</span><span
 class="pln">gff</span><span class="pun">/</span></pre>
<p>Concatenated all of the gff files in this directory into a single
gff output file. </p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cat </span><span class="pun">*.</span><span
 class="pln">gff </span><span class="pun">&gt;</span><span class="pln"> HEX0014K09</span><span
 class="pun">.</span><span class="pln">gff</span></pre>
<h2>3. Convert concatenated gff file to game.xml format</h2>
<p>Converting the concatenated gff file can be done using the
batch_convert.pl progra. This takes as its input the input directory of
fasta files. The output will be stored in the game directory. A copy of
all of the game.xml files will also be placed in the base
wheat_analysis directory.<br>
</p>
<pre class="prettyprint"><span class="pln"> &nbsp;</span><span
 class="pun">&gt;</span><span class="pln">cd </span><span class="pun">/</span><span
 class="pln">home</span><span class="pun">/</span><span class="pln">YourName</span><span
 class="pun">/</span><span class="pln">projects</span><span class="pun">/</span><span
 class="pln">wheat_annotation</span><span class="pun">/</span><span
 class="pln">wheat_analysis<br>&nbsp; </span><span class="pun">&gt;</span><span
 class="pln">batch_convert</span><span class="pun">.</span><span
 class="pln">pl </span><span class="pun">-</span><span class="pln">i wheat_analysis</span><span
 class="pun">/</span><span class="pln">masked_soft</span><span
 class="pun">/ -o </span><span class="pln">wheat_analysis</span><span
 class="pun">/</span></pre>
<hr style="width: 100%; height: 2px;">
<h1><a name="human_curation"></a>G. Human Curation of Computational
Results<br>
</h1>
The Apollo program is used for human curation of the computational
results generated above.<br>
<br>
<br>
<hr style="width: 100%; height: 2px;">
<h1><a name="curation_quality_control"></a>Curation Quality Control<br>
</h1>
There are scripts available for quality control and feature comparison.<br>
<br>
Convert game.xml annotations to GFF format.<br>
<br>
Venn Diagram of sequence features<br>
<br>
Predict Function of Human Curated Gene Models<br>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="additional_information"></a>IV. ADDITIONAL INFORMATION</h1>
The following information is provided for additional reference.<br>
<br>
<hr style="width: 100%; height: 2px;"><br>
<h1><a name="apollo_data_tiers"></a>1. Apollo Data Tiers</h1>
The GFF files produced by the computational results above are intended
to be visualized using the Apollo Genome Annotation Curation Tool. The
way that Apollo displays data can be modified through the use of tiers
files and style files. These files are placed in your home directory in
the directory named
'.apollo'. The dot in the front of the name indicates that this
direcotry is hidden.A
tiers file and a style file has been created for visualizing the
results for wheat annotation. These are included as part of the
DAWG-PAWS SVN package, but are also available at:<br>
<ul>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/dawgpaws/trunk/apollo/wheat.tiers">wheat.tiers</a></li>
  <li><a
 href="http://dawgpaws.svn.sourceforge.net/viewvc/dawgpaws/trunk/apollo/wheat.style">wheat.style</a><br>
  </li>
</ul>
I prefer to point to these using symlinks from within my home directory
as shown below.<br>
<pre class="prettyprint"><span class="pln"> </span><span class="pun"> &gt;</span><span
 class="pln">cd </span><span class="pun">$HOME</span><span class="pln">/.apollo </span><span
 class="pun"><br>  &gt;</span><span class="pln">ln -s $HOME/apps/dawgpaws/apollo/wheat.tiers fly.tiers</span><br><span
 class="pun">  &gt;</span><span class="pln">ln -s $HOME/apps/dawgpaws/apollo/wheat.style fly.style</span><span
 class="pun"></span></pre>
These will create files name fly.tiers and fly.style that just point to
the location of the wheat.tiers and wheat.style files in the dawgpaws
SVN directory. The advantage of this is that when additional data tiers
are added to the computational evidences that I use, I can create a new
tiers file and propagate this to all of the annotators using SVN.<br>
<br>
<hr style="width: 100%; height: 2px;">
<h1><a name="directory_structure"></a>Computational Results Directory
Structure</h1>
The DAWG-PAWS programs will store the computational results in a
predefined directory structure. The software will establish this
structure when output from the various annotation programs are produced
or parsed. <br>
<br>
An example below shows an example directory structure of computational
results for the BAC identified as HEX0014K09.<br>
<ul>
  <li><strong>HEX0014K09</strong> </li>
  <ul>
    <li><strong>blast</strong> - <i>Results of blast searches</i> </li>
    <li><strong>find_ltr</strong> - <i>Results from the find_ltr
program.</i> </li>
    <li><strong>findmite</strong> - <i>Results from the FINDMITE
program</i> </li>
    <li><strong>game</strong> - <i>game.xml files</i> </li>
    <li><strong>gff</strong> - <i>gff format data files</i> </li>
    <li><strong>hmmer</strong> - <i>Results from the hmmsearch program.</i>
    </li>
    <ul>
      <li><strong>rice_mite</strong> - <i>Rice MITE hmm profiles</i> </li>
      <li><strong>rice_mule</strong> - <i>Rice MULE hmm profiles</i> </li>
      <li><strong>wheat_ltr</strong> - <i>Wheat LTR hmm profiles</i> </li>
    </ul>
    <li><strong>ltr_finder</strong> - <i>Results from the ltr_finder
program.</i> </li>
    <li><strong>ltr_seq</strong> - <i>ltr_seq output</i> </li>
    <li><strong>ltr_struc</strong> - <i>Results from the ltr_struc
program.</i> </li>
    <li><strong>repseek</strong> - <i>Results from the Repseek program</i>
    </li>
    <li><strong>rm</strong> - <i>Repeatmasker results</i> </li>
    <li><strong>ta</strong> - <i>TriAnnotation results</i></li>
    <li><strong>tenest</strong> - <i>Results from TE NEST</i></li>
  </ul>
</ul>
<hr style="width: 100%; height: 2px;">
<h1><a name="manfiles"></a>Instructions for Individual Programs<br>
</h1>
Links to the program manuals for the indvidual scripts are listed below
in
alphabetical
order. These are essentialy extensive help files for each command line
program use in the genome annotation process. <br>
<dl>
  <dt style="font-weight: bold;"><a href="man/batch_blast.html">batch_blast.pl</a></dt>
  <dd>Do NCBI-BLAST searches for a set of fasta files</dd>
  <dt style="font-weight: bold;"><a href="man/batch_cnv_blast2gff.html">batch_cnv_blast2gff.pl</a></dt>
  <dd>Convert blast output to GFF format</dd>
  <dt style="font-weight: bold;"><a href="man/batch_cnv_ta2ap.html">batch_cnv_ta2ap.pl</a></dt>
  <dd>Convert TriAnnotation to Apollo GFF</dd>
  <dt>batch_eugene</dt>
  <dd>Run the eugene annotation program in batch modes<br>
  </dd>
  <dt style="font-weight: bold;"><a href="man/batch_findgaps.html">batch_findgaps.pl</a></dt>
  <dd>Annotate gaps in a fasta file</dd>
  <dt style="font-weight: bold;"><a href="man/batch_findltr.html">batch_findltr.pl</a></dt>
  <dd>Run the find_ltr.pl program in batch mode</dd>
  <dt style="font-weight: bold;"><a href="man/batch_findmite.html">batch_findmite.pl</a></dt>
  <dd>Run the findmite program in batch mode</dd>
  <dt style="font-weight: bold;"><a href="man/batch_game2gff.html">batch_game2gff.pl</a></dt>
  <dd>Convert game.xml annotations to GFF format<br>
  </dd>
  <dt style="font-weight: bold;"><a href="man/batch_genemark.html">batch_genemark.pl</a></dt>
  <dd>Run GeneMark.hmm and parse results to a GFF format file</dd>
  <dt style="font-weight: bold;"><a href="man/batch_genscan.html">batch_genscan.pl</a></dt>
  <dd>Run genscan in batch mode and parse results to GFF format</dd>
  <dt style="font-weight: bold;">batch_gff2game.pl</dt>
  <dd>Convert GFF files to the game.xml format</dd>
  <dt style="font-weight: bold;"><a href="man/batch_hardmask.html">batch_hardmask.pl</a></dt>
  <dd>Hardmask a directory of softmasked fasta files</dd>
  <dt style="font-weight: bold;"><a href="man/batch_hmmer.html">batch_hmmer.pl</a></dt>
  <dd>Run the HMMER program in batch mode</dd>
  <dt style="font-weight: bold;"><a href="man/batch_ltrfinder.html">batch_ltrfinder.pl</a></dt>
  <dd>Run the LTRFinder program in batch mode</dd>
  <dt>batch_ltrseq.pl</dt>
  <dd>Run the LTR_seq program in batch mode<br>
    <span style="font-style: italic;">This program is currently in
development</span><br>
  </dd>
  <dt style="font-weight: bold;"><a href="man/batch_mask.html">batch_mask.pl</a></dt>
  <dd>Run RepeatMasker and parse results to a GFF format file</dd>
  <dt style="font-weight: bold;"><a href="man/batch_seq_summary.html">batch_seq_summary.pl</a></dt>
  <dd>Print summary info for a directory of sequence files</dd>
  <dt>batch_tenest.pl</dt>
  <dd>Run the TE nest program in batch mode on a directory of fasta
files.<br>
  </dd>
  <dt style="font-weight: bold;"><a href="man/clust_write_shell.html">clust_write_shell.pl</a></dt>
  <dd>Write shell scripts for the Platform LSF
queuing system</dd>
  <dt>cnv_fgenesh2gff.pl</dt>
  <dd>Convert a single FGENESH output to GFF format</dd>
  <dt>cnv_findltr2gff.pl</dt>
  <dd>Convert a single findltr output file to GFF format</dd>
  <dt>cnv_findmite2gff.pl</dt>
  <dd>Convert a single findmite output file to GFF format</dd>
  <dt>cnv_genemark2gff.pl</dt>
  <dd>Convert a single output file from genemark to GFF format</dd>
  <dt>cnv_gff2game.pl</dt>
  <dd>Convert a GFF file to the game.xml format.</dd>
  <dt>cnv_ltrfinder2gff.pl</dt>
  <dd>Convert a single output file from ltrfinder to GFF format</dd>
  <dt style="font-weight: bold;"><a href="man/cnv_ltrseq2gff.html">cnv_ltrseq2gff.pl</a></dt>
  <dd>Convert a single output file from LTR_seq to GFF format</dd>
  <dt style="font-weight: bold;"><a href="man/cnv_ltrstruc2gff.html">cnv_ltrsruc2gff.pl</a></dt>
  <dd>Convert output from LTR_struc to GFF format<br>
  </dd>
  <dt>cnv_repmask2gff.pl</dt>
  <dd>Convert a single output file from RepeatMasker to GFF format</dd>
  <dt style="font-weight: bold;"><a href="man/cnv_ta2ap.html">cnv_ta2ap.pl<br>
    </a></dt>
  <dd>Convert the TriAnnotation GFF3 output to apollo compatible GFF
format</dd>
  <dt>cnv_tenest2gff.pl</dt>
  <dd>Convert a single output from the TE Nest to GFF format</dd>
  <dt style="font-weight: bold;"><a href="man/dir_merge.html">dir_merge.pl</a></dt>
  <dd>Merge directories of DAWG-PAWS output to a single directory</dd>
  <dt style="font-weight: bold;"><a href="man/fasta_dirsplit.html">fasta_dirsplit.pl</a></dt>
  <dd>Split a directry of fasta files into a n subdirectories</dd>
  <dt style="font-weight: bold;"><a href="man/fasta_shorten.html">fasta_shorten.pl</a></dt>
  <dd>Change fasta headers to shorter names</dd>
  <dt>fasta_trep_shorten.pl</dt>
  <dd>Change fasta headers in a TREP fasta file to shorter names</dd>
  <dt style="font-weight: bold;"><a href="man/fetch_tenest.html">fetch_tenest.pl</a></dt>
  <dd>Download a set of results from the TE Nest web server&nbsp;</dd>
  <dt style="font-weight: bold;"><a href="man/ltrstruc_prep.html">ltrstruc_prep.pl</a></dt>
  <dd>Create the files needed to run the LTR_struc program</dd>
  <dt>push_files.pl</dt>
  <dd>Push a set of files from the local machine to a set of machines
using SFTP</dd>
  <dt>run_fgenesh.pl</dt>
  <dd>Run fgenesh in the softberry interface</dd>
  <dt>run_genscan.pl</dt>
  <dd>Run the genescan program</dd>
  <dt style="font-weight: bold;"><a href="man/seq_oligocount.html">seq_oligocount.pl</a></dt>
  <dd>Count oligo redundancy for a input sequence<br>
  </dd>
  <dt>vennseq.pl</dt>
  <dd>Create Venn Diagrams of sequence features<span
 style="font-weight: bold;"></span></dd>
  <dt><br>
  </dt>
</dl>
<hr style="width: 100%; height: 2px;">
<h1><a name="literature_cited"></a>LITERATURE CITED<br>
</h1>
Achaz, G., F. Boyer, et al. (2007). "Repseek, a tool to retrieve
approximate repeats from large DNA sequences." Bioinformatics 23(1):
119-21.<br>
<br>
Altschul, S. F., T. L. Madden, et al. (1997). "Gapped BLAST and
PSI-BLAST: a new generation of protein database search programs."
Nucleic Acids Res 25(17): 3389-402.<br>
<br>
Benson, G. (1999). "Tandem repeats finder: a program to analyze DNA
sequences." Nucleic Acids Res 27(2): 573-80.<br>
<br>
Burge, C. and S. Karlin (1997). "Prediction of complete gene structures
in human genomic DNA." J Mol Biol 268(1): 78-94.<br>
<br>
Edgar, R. C. and E. W. Myers (2005). "PILER: identification and
classification of genomic repeats." Bioinformatics 21 Suppl 1: i152-8.<br>
<br>
Juretic, N., T. E. Bureau, et al. (2004). "Transposable element
annotation of the rice genome." Bioinformatics 20(2): 155-160.<br>
<br>
Kalyanaraman, A. and S. Aluru (2006). "Efficient algorithms and
software for detection of full-length LTR retrotransposons." J
Bioinform Comput Biol 4(2): 197-216.<br>
<br>
Kent, W. J. (2002). "BLAT--the BLAST-like alignment tool." Genome Res
12(4): 656-64.<br>
<br>
Kurtz, S. (2004). Vmatch. <a href="http://www.vmatch.de/">http://www.vmatch.de/</a><br>
<br>
Kronmiller, B. A. and R. P. Wise (2007). "TE nest: Automated
chronological annotation and visualization of nested plant transposable
elements." Plant Physiol.<br>
<br>
Lewis, S. E., S. M. Searle, et al. (2002). "Apollo: a sequence
annotation editor." Genome Biol 3(12): RESEARCH0082.<br>
<br>
McCarthy, E. M. and J. F. McDonald (2003). "LTR_STRUC: a novel search
and identification program for LTR retrotransposons." Bioinformatics
19(3): 362-7.<br>
<br>
Parra, G., E. Blanco, et al. (2000). "GeneID in Drosophila." Genome Res
10(4): 511-5.<br>
<br>
Quesneville, H., C. M. Bergman, et al. (2005). "Combined evidence
annotation of transposable elements in genome sequences." PLoS Comput
Biol 1(2): 166-75.<br>
<br>
Kurtz, S., J. V. Choudhuri, et al. (2001). "REPuter: the manifold
applications of repeat analysis on a genomic scale." Nucleic Acids Res
29(22): 4633-42.<br>
<br>
Rho, M., J. H. Choi, et al. (2007). "De novo identification of LTR
retrotransposons in eukaryotic genomes." BMC Genomics 8: 90.<br>
<br>
Schiex, T., A. Moisan, et al. (2001). EuGene: An Eucaryotic Gene Finder
that combines several sources of evidence. . Computational Biology. O.
Gascuel and M.-F. Sagot: 111-125.<br>
<br>
Tu, Z. (2001). "Eight novel families of miniature inverted repeat
transposable elements in the African malaria mosquito, Anopheles
gambiae." Proc Natl Acad Sci U S A 98(4): 1699-704.<br>
<br>
Wicker, T., F. Sabot, et al. (2007). "A unified classification system
for eukaryotic transposable elements." Nat Rev Genet 8(12): 973-82.<br>
<br>
Xu, Z. and H. Wang (2007). "LTR_FINDER: an efficient tool for the
prediction of full-length LTR retrotransposons." Nucleic Acids Res
35(Web Server issue): W265-8.<br>
<br>
<hr>
<table style="border-collapse: collapse;" id="AutoNumber2" border="0"
 bordercolor="#111111" cellpadding="0" cellspacing="0" width="100%">
  <tbody>
    <tr>
      <td width="50%"><i>Author: <a
 href="http://www.arches.uga.edu/%7Ejestill">James Estill</a><br>
Last Updated: Nov 10, 2008</i></td>
      <td width="50%"> <br>
      </td>
      <td><a href="http://sourceforge.net/"> <img
 src="http://sourceforge.net/sflogo.php?group_id=204962&amp;type=5"
 alt="SourceForge.net Logo" border="0" height="62" width="210"></a> </td>
    </tr>
  </tbody>
</table>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<script src="http://www.google-analytics.com/urchin.js"
 type="text/javascript">
</script>
<script type="text/javascript">
_uacct = "UA-2207628-5";
urchinTracker();
</script>
</body>
</html>
